kafka-connect-s3/src/main/java/io/confluent/connect/s3/hooks/BlockingKafkaPostCommitHook.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Add here
+ */
+
+package io.confluent.connect.s3.hooks;
+
+import io.confluent.connect.s3.S3SinkConnectorConfig;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.errors.AuthorizationException;
+import org.apache.kafka.common.errors.OutOfOrderSequenceException;
+import org.apache.kafka.common.errors.ProducerFencedException;
+import org.apache.kafka.common.errors.UnsupportedVersionException;
+import org.apache.kafka.common.header.internals.RecordHeader;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.kafka.connect.errors.RetriableException;
+
+import java.util.Properties;
+import java.util.Set;
+
+public class BlockingKafkaPostCommitHook implements PostCommitHook {
+
+  private static final Logger log = LoggerFactory.getLogger(BlockingKafkaPostCommitHook.class);
+  private String kafkaTopic;
+  private KafkaProducer<String, String> kafkaProducer;
+
+  @Override
+  public void init(S3SinkConnectorConfig config) {
+    if (kafkaProducer != null) {
+      close();
+    }
+    kafkaTopic = config.getPostCommitKafkaTopic();
+    kafkaProducer = newKafkaPostCommitProducer(config);
+    log.info("BlockingKafkaPostCommitHook initialized successfully");
+  }
+
+  @Override
+  public void put(Set<String> s3ObjectPaths) {
+    try {
+      kafkaProducer.beginTransaction();
+      log.info("Transaction began");
+
+      for (String s3ObjectPath : s3ObjectPaths) {
+        ProducerRecord<String, String> record = new ProducerRecord<>(kafkaTopic, s3ObjectPath);
+        record.headers().add(new RecordHeader("accountId", "300".getBytes()));
```

Comment:
what is "300" ? it's account specific also need to add start time of the file

Reply:
This was just a test and I have committed it by mistake. Will add the account Id and Timestamp in a different PR.

Reply:
ok